syntax = "proto3";

package spectrogramgenerator;

service SpectrogramGenerator {
    rpc GenerateSpectrogram (SpectrogramGeneratorRequest) returns (SpectrogramGeneratorResponse);
}

message SpectrogramGeneratorRequest {
    string window_type = 1;
    int32 n_samples = 2;
    float frequency_cutoff = 3;
    int32 spectrogram_min = 4;
    string uri_to_wav = 5;
}

message SpectrogramGeneratorResponse {
    string message = 1;
    string uri_to_png = 2;
}